hack/wait-until-hpa-ready/main.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+func main() {
+	var hpaName string
+	var timeout int
+	var kubeconfigPath string
+
+	defaultKubeconfigPath := filepath.Join(homedir.HomeDir(), ".kube", "config")
+
+	pflag.IntVar(&timeout, "timeout", 600, "The timeout for the check.")
+	pflag.StringVar(&hpaName, "hpa", "", "HPA to check")
+	pflag.StringVar(&kubeconfigPath, "kubeconfig-path", defaultKubeconfigPath, "Absolute path to the KubeconfigPath file")
+	pflag.Parse()
+
+	if len(hpaName) == 0 {
+		fmt.Println("hpa flag is mandatory")
+		os.Exit(1)
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		fmt.Println("Error reading the kubeconfig:", err.Error())
+		os.Exit(1)
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	namespace, err := client.CoreV1().Namespaces().Get(context.Background(), os.Getenv("NAMESPACE"), metav1.GetOptions{})
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	hpaClientV2 := client.AutoscalingV2().HorizontalPodAutoscalers(namespace.Name)
+	hpaClientV1 := client.AutoscalingV1().HorizontalPodAutoscalers(namespace.Name)
+
+	pollInterval := time.Second
+
+	err = wait.Poll(pollInterval, 0, func() (done bool, err error) {
+		hpav2, err := hpaClientV2.Get(
+			context.Background(),
+			hpaName,
+			metav1.GetOptions{},
+		)
+		if err != nil {
+			hpav1, err := hpaClientV1.Get(
+				context.Background(),
+				hpaName,
+				metav1.GetOptions{},
+			)
+			if err != nil {
+				fmt.Println("HPA", hpaName, "not found")
+				return false, nil
+			}
+
+			if hpav1.Status.CurrentCPUUtilizationPercentage == nil {
+				fmt.Println("Current metrics are not set yet for HPA", hpaName)
+				return false, nil
+			}
+			return true, nil
+		}
+
+		if hpav2.Status.CurrentMetrics == nil {
+			fmt.Println("Current metrics are not set yet for HPA", hpaName)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	fmt.Println(hpaName, "is ready!")
```

Comment:
personally, I haven't seen `fmt.Println` to be used this way. Consider using `fmt.Printf`